### ISSUE TITLE
Unreleased "Profile" removal in aws-partitions gem

### ISSUE THREAD

**Issue by robbkidd** [aws/aws-sdk-ruby#2446]
- [x] I've gone though [Developer Guide for v3](https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-ruby) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-ruby/issues) and didn't find any solution
  
**Describe the bug**

There is a currently unreleased [code change](https://github.com/aws/aws-sdk-ruby/commit/0ed223e914d10abffe29eee8e7542c4ce97916fa#diff-f6072f5537ff2d5b76e149332bde78a7a00f9b9cc22d1b9526ff7bbda7caa0ed) in the `aws-partitions` gem. I suspect this might be due to the gem's version not being revved in that commit ([0ed223e9](https://github.com/aws/aws-sdk-ruby/commit/0ed223e914d10abffe29eee8e7542c4ce97916fa#diff-f6072f5537ff2d5b76e149332bde78a7a00f9b9cc22d1b9526ff7bbda7caa0ed)) or in [the commits merged since](https://github.com/aws/aws-sdk-ruby/compare/aws:0ed223e...aws:b2afef1).

**Gem name:**

aws-partitions v1.400.0 - v1.402.0

**Version of Ruby, OS environment**

Ruby 2.2 - 2.7

**To Reproduce (observed behavior)**
Steps to reproduce the behavior (please share code or minimal repo)

```
» gem list | grep aws-partitions
aws-partitions (1.402.0)
» irb
irb(main):001:0> require 'aws-sdk'
=> true
irb(main):002:0> Aws::Partitions.service_ids.keys.map do |service|
irb(main):003:1*    Aws.const_get(service)
irb(main):004:1> end
Traceback (most recent call last):
        7: from /Users/robbkidd/.asdf/installs/ruby/2.6.3/bin/irb:23:in `<main>'
        6: from /Users/robbkidd/.asdf/installs/ruby/2.6.3/bin/irb:23:in `load'
        5: from /Users/robbkidd/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):2
        3: from (irb):2:in `map'
        2: from (irb):3:in `block in irb_binding'
        1: from (irb):3:in `const_get'
NameError (uninitialized constant Aws::Profile)
```

Error appeared in the nightly [the Honeycomb beeline-ruby integration tests](https://github.com/honeycombio/beeline-ruby/blob/c318943d1eeebf1a12146c828d8d469d7c5db435/spec/honeycomb/integrations/aws_spec.rb#L43) starting on Dec 2 and continuing to today. The integration tests walk the collection of `service_ids` declared in `Aws::Partitions` and try to access the associated classes. Over the course of a few versions of the `aws-partitions` gem (1.400.0-1.402.0), `Profile` was removed from the gem to be added under a different name. 0ed223e9 appears to have one more removal left to release.

**Comment by mullermp** [comment]
Thank you for reporting this. We thought we cleaned up everything around this. Long story short, an AWS service launched with "Profile" and it caused conflicts with CLI so SDK teams were forced to remove it and it was re-released as CustomerProfiles instead.

**Comment by mullermp** [comment]
Github 'master' branch appears to be in good state. A new release is going out today with partitions changes, so I think that this should get resolved once that finishes.

**Comment by robbkidd** [comment]
Much love sent y'all's way. Totally understand how much fiddly work it is to undo code generation. Thanks for the quick response!

**Comment by mullermp** [comment]
This appears to be fixed with 1.403.0. Re-open if it's still a problem for you.
